File: examples/php/email_validator_atom.php
<?php
/**
 * Atom: validar formato de email en PHP.
 *
 * Devuelve un arreglo con claves `valid` y `error` para mantener compatibilidad con
 * las composiciones declarativas descritas en AADM 2.0.
 */
function validate_email_atom(string $email): array
{
    $valid = filter_var($email, FILTER_VALIDATE_EMAIL) !== false;
    return [
        'valid' => $valid,
        'error' => $valid ? null : 'Invalid format',
    ];
}

File: examples/python/email_validator.atom.py
"""Atom: validate email against RFC-like regex.

Pequeño propósito: comprobar formato y devolver resultado estructurado.
"""
import re

EMAIL_PATTERN = re.compile(r"^[^@\s]+@[^@\s]+\.[^@\s]+$")


def validate_email(email: str) -> dict:
    """Valida el correo y describe el resultado.

    Retorna estructura con `valid` boolean y `error` opcional para
    facilitar composición declarativa.
    """

    valid = bool(EMAIL_PATTERN.match(email))
    return {"valid": valid, "error": None if valid else "Invalid format"}
